Sources/NIOPosix/NonBlockingFileIO.swift

```diff
@@ -712,3 +730,356 @@ public struct NIODirectoryEntry: Hashable {
     }
 }
 #endif
+
+@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
```

Comment:
Super nit: But can we move those to the individual methods I always feel like it is easier to discover the availability then